packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -740,3 +741,40 @@ const tryWrappingBlockStatementInIIFE = (node: t.BlockStatement | t.Expression)
  */
 export const pickFunctionBody = (node: t.Function): t.Expression =>
   tryWrappingBlockStatementInIIFE(node.body);
+
+/**
+ * Get comments for `path` in both the line before and on the same line.
+ *
+ * e.g.
+ * `<div css={{color: 'green'}} /> // @compiled-disable` will output `{before: [], same: [...]}
+ *
+ * @param path {NodePath<t.Node>}
+ * @param meta {Metadata} Context for the transform
+ * @returns {before: t.CommentLine[], same: t.CommentLine[]} Comments before and on the same line as the input path
+ */
+export const getNodeDirectiveComments = (
```

Comment:
Can this go in its own util file? The large utils are unbearable to read. Maybe something like `utils/directives.ts` though I'm confused why `directive` is needed in the name since it's meant to be generically retrieving comments? Maybe just `utils/comments.ts` and `getComments`?

Reply:
Yeah makes sense to me. I put directive as the function only retrieves the above and same line, which is close to the directive use-case.

Reply:
I went with `getNodeComments` as I want to keep the context that the comments are relevant to the AST node specified